kafka.go

```diff
@@ -213,13 +213,31 @@ func (client *KafkaClient) Publish(publishBuilder *PublishBuilder) error {
 
 // Publish send event to a topic
 func (client *KafkaClient) publishEvent(ctx context.Context, topic, eventName string, config kafka.WriterConfig,
-	message kafka.Message) error {
+	message kafka.Message) (err error) {
+
+	var writer *kafka.Writer
+	defer func() {
+		if r := recover(); r != nil {
+			logrus.Warnf("unable to publish event to: %v. attempt to recover", topic)
+			if writer == nil {
+				err = errors.New("writer is nil")
+				return
+			}
+			err = writer.Close()
+			if err != nil {
+				logrus.Errorf("unable to publish event to: %v, unable to close writer: %v", topic, err)
+			}
+			client.deleteWriter(config.Topic)
+			err = fmt.Errorf("recover: %v", r)
```

Comment:
dont need this error

Reply:
the error here is to enable retrying publish event

Reply:
got it